indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -572,7 +638,7 @@ private static boolean isGuaranteedRollup(IndexIOConfig ioConfig, IndexTuningCon
    *
    * @return generated {@link ShardSpecs} representing a map of intervals and corresponding shard specs
    */
-  private ShardSpecs determineShardSpecs(
+  private Map<Interval, Pair<ShardSpecFactory, Integer>> determineShardSpecs(
```

Comment:
Hmm, the javadoc link on the return type is no longer correct here (and a couple of other places). However, there might be enough `Map<Interval, Pair<ShardSpecFactory, Integer>>` to consider capturing it into a type similar to `ShardSpecs`?

Reply:
Nice finding. Fixed the javadoc.